prov/sockets/src/sock_ep_msg.c

```diff
@@ -296,7 +296,7 @@ static int sock_pep_create_listener(struct sock_pep *pep)
 	for (p = s_res; p; p = p->ai_next) {
 		pep->cm.sock = socket(p->ai_family, p->ai_socktype,
 				     p->ai_protocol);
-		if (pep->cm.sock) {
+		if (pep->cm.sock > 0) {
```

Comment:
Theoretically, application can close fd 0 (stdin) and 0 can be returned here. If the purpose is check for errors, then 0 is not an error case.